internal/controllers/clusterextension_controller.go

```diff
@@ -409,6 +411,23 @@ func (r *ClusterExtensionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&ocv1alpha1.ClusterExtension{}).
 		Watches(&catalogd.Catalog{},
 			handler.EnqueueRequestsFromMapFunc(clusterExtensionRequestsForCatalog(mgr.GetClient(), mgr.GetLogger()))).
+		WithEventFilter(predicate.Funcs{
+			UpdateFunc: func(ue event.UpdateEvent) bool {
+				oldObject, isOldCatalog := ue.ObjectOld.(*catalogd.Catalog)
+				newObject, isNewCatalog := ue.ObjectNew.(*catalogd.Catalog)
+
+				if !isOldCatalog || !isNewCatalog {
+					return true
+				}
+
+				if oldObject.Status.ResolvedSource != nil && newObject.Status.ResolvedSource != nil {
+					if oldObject.Status.ResolvedSource.Image != nil && newObject.Status.ResolvedSource.Image != nil {
+						return oldObject.Status.ResolvedSource.Image.Ref != newObject.Status.ResolvedSource.Image.Ref
```

Comment:
IIRC the `.status.resolvedSource.image.resolvedRef` is probably what we want here. See https://github.com/operator-framework/catalogd/blob/29870e83a0dd4b1d99d69796a3d4c61887737f0d/api/core/v1alpha1/catalog_types.go#L120-L121

Reply:
Thanks for pointing out @everettraven! Didn't know that existed.